test/box-luatest/gh_9823_net_replicaset_test.lua

```diff
@@ -180,3 +180,238 @@ g.test_net_replicaset_basics = function(cg)
 
     t.assert_equals(leader_name, new_leader_name)
 end
+
+-- Test watch_leader method.
+g.test_watch_leader = function(cg)
+    local config = test_config()
+    local cluster = cluster.new(cg, config)
+    cluster:start()
+
+    -- Set initial value to event 'key1'.
+    cluster:each(function(server)
+        if server.alias:startswith('storage') then
+            server:exec(function()
+                if not box.info.ro then
+                    box.broadcast('key1', 1)
+                else
+                    box.broadcast('key1', 2)
+                end
+            end)
+        end
+    end)
+
+    -- Subscribe and check that event was received only from leader.
+    local leader_name = cluster.router:exec(function()
```

Comment:
I tried to do this way, spend several hours debugging and found out that the test cluster is started in a random temporary folder and I found no public API to retrieve its name. There's `cluster._dir` but it seems to be private and a subject to change.
Since the cluster instances are started with an unix socket listen port, they are cannot be accessed from the main test process.

Reply:
I mean now the test uses static relative unix socket paths. Perhaps we could collect connect_cfg from test.config-luatest.cluster. Is it worth it?

Reply:
Ok, remade in this way.